synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go

```diff
@@ -308,23 +309,39 @@ func (s *ProcessorTrustedBatchSync) GetModeForProcessBatch(trustedNodeBatch *typ
 			}
 		}
 	}
+
 	if result.Mode == "" {
-		return result, fmt.Errorf("failed to get mode for process batch %v", trustedNodeBatch.Number)
+		return result, fmt.Errorf("batch %v: failed to get mode for process ", trustedNodeBatch.Number)
 	}
+
 	result.BatchNumber = uint64(trustedNodeBatch.Number)
 	result.StateBatch = stateBatch
 	result.TrustedBatch = trustedNodeBatch
 	result.PreviousStateBatch = statePreviousBatch
 	result.OldAccInputHash = statePreviousBatch.AccInputHash
 	result.Now = s.timeProvider.Now()
 	result.DebugPrefix = fmt.Sprintf("%s mode %s:", debugPrefix, result.Mode)
+
+	if isTrustedBatchEmptyAndClosed(trustedNodeBatch) {
+		if s.Cfg.AcceptEmptyClosedBatches {
+			log.Infof("%s Batch %v: TrustedBatch Empty and closed, accepted due configuration", result.DebugPrefix, trustedNodeBatch.Number)
+		} else {
+			log.Infof("%s Batch %v: TrustedBatch Empty and closed, rejected due configuration", result.DebugPrefix, trustedNodeBatch.Number)
+			return result, fmt.Errorf("%s Batch %v: TrustedBatch Empty and closed, rejected due configuration", result.DebugPrefix, trustedNodeBatch.Number)
```

Comment:
shouldn't we return `nil` instead of `result` here?

Reply:
The returned type is not a pointer, so is just to return something